Sources/DataConnect.swift

```diff
@@ -170,7 +187,7 @@ private class InstanceStore {
     autoreleaseFrequency: .workItem
   )
 
-  var instances = [InstanceKey: DataConnect]()
+  private var instances = [InstanceKey: DataConnect]()
```

Comment:
This is unrelated to this PR and likely not very important in practice, but I had learned that Swift dictionaries strongly reference their values. So once an instance is added to the dictionary, it will never be deallocated. To work around this, you can box the value in a type that holds a weak reference to it. This way, the instance cache will not hold a strong reference and prevent it from be deallocated. Again though, likely not critical since there likely won't be many instances created and they likely won't be deallocated often/ever during lifetime of app.

Example of dictionary type
https://github.com/firebase/firebase-ios-sdk/blob/dd0a4c6cb84aace05e8e49469f0fb2c1f70fd155/FirebaseCore/Internal/Sources/HeartbeatLogging/HeartbeatStorage.swift#L61-L63

Example of removing an instance from the dictionary upon deallocation
https://github.com/firebase/firebase-ios-sdk/blob/dd0a4c6cb84aace05e8e49469f0fb2c1f70fd155/FirebaseCore/Internal/Sources/HeartbeatLogging/HeartbeatStorage.swift#L110

Example of weak container type
https://github.com/firebase/firebase-ios-sdk/blob/dd0a4c6cb84aace05e8e49469f0fb2c1f70fd155/FirebaseCore/Internal/Sources/HeartbeatLogging/WeakContainer.swift

Reply:
Good to know. I'll track this for future though in reality we won't see many instances per SDK linkage. There should in theory be just one per connector. So even with multi connector apps, might not be more than single digits.

Reply:
+1, agreed